asciidoc.py

```diff
@@ -2265,7 +2302,13 @@ def gen_id(title):
         base_id = re.sub(r'\W+', '_', title).strip('_').lower()
         if 'ascii-ids' in document.attributes:
             # Replace non-ASCII characters with ASCII equivalents.
-            base_id = unicodedata.normalize('NFKD', base_id).encode('ascii', 'ignore').decode('ascii')
+            normalized_id = ''
+            for part in unicodedata.normalize('NFKD', base_id):
+                if unicodedata.category(part) == 'Mn':
+                    continue
+                for character in part:
+                    normalized_id += NON_NFKD_MAP.get(character, character)
+            base_id = normalized_id.encode('ascii', 'ignore').decode('ascii')
```

Comment:
I am wondering if we really need this - the previous function seems to handle all the issues. In previous implementation the NFKD method decomposes the character to base letter and accents. Then encode('ascii', 'ignore') helped us to get rid of accents. Now we do it in line 2307.
Am I correct or am I missing something?